test/Microsoft.AspNet.Razor.Runtime.Test/DefaultTagHelperContentTest.cs

```diff
@@ -0,0 +1,201 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.IO;
+using Xunit;
+
+namespace Microsoft.AspNet.Razor.Runtime
+{
+    public class DefaultTagHelperContentTest
+    {
+        [Fact]
+        public void CanSetContent()
+        {
+            // Arrange
+            var tagHelperContent = new DefaultTagHelperContent();
+            var expected = "Hello World!";
+
+            // Act
+            tagHelperContent.SetContent(expected);
+
+            // Assert
+            Assert.Equal(expected, tagHelperContent.ToString());
+        }
+
+        [Fact]
+        public void SetContentClearsExistingContent()
+        {
+            // Arrange
+            var tagHelperContent = new DefaultTagHelperContent();
+            var expected = "Hello World!";
+
+            // Act
+            tagHelperContent.SetContent("Contoso");
+            tagHelperContent.SetContent(expected);
+
+            // Assert
+            Assert.Equal(expected, tagHelperContent.ToString());
+        }
+
+        [Fact]
+        public void CanGetContent()
+        {
+            // Arrange
+            var tagHelperContent = new DefaultTagHelperContent();
+            var expected = "Hello World!";
+            tagHelperContent.SetContent(expected);
+
+            // Act
+            var actual = tagHelperContent.GetContent();
+
+            // Assert
+            Assert.Equal(expected, actual);
+        }
+
+        [Fact]
+        public void CanAppendContent()
+        {
+            // Arrange
+            var tagHelperContent = new DefaultTagHelperContent();
+            var expected = "Hello World!";
+
+            // Act
+            tagHelperContent.Append(expected);
+
+            // Assert
+            Assert.Equal(expected, tagHelperContent.GetContent());
+        }
+
+        [Fact]
+        public void CanIdentifyIfModified()
+        {
+            // Arrange
+            var tagHelperContent1 = new DefaultTagHelperContent();
+            var tagHelperContent2 = new DefaultTagHelperContent();
+            var tagHelperContent3 = new DefaultTagHelperContent();
+
+            // Act
+            tagHelperContent1.SetContent("Test");
+            tagHelperContent2.Append("Foo");
+            tagHelperContent3.Clear();
+
+            // Assert
+            Assert.True(tagHelperContent1.IsModified);
+            Assert.True(tagHelperContent2.IsModified);
+            Assert.True(tagHelperContent3.IsModified);
+        }
+
+        [Fact]
+        public void CanAddWhiteSpaces()
+        {
+            // Arrange
+            var tagHelperContent = new DefaultTagHelperContent();
+
+            // Act
+            tagHelperContent.SetContent("  ");
+
+            // Assert
+            Assert.Equal(2, tagHelperContent.ToString().Length);
+        }
+
+        [Fact]
+        public void CanIdentifyWhiteSpaces()
```

Comment:
also confirm that tabs, newlines, and returns don't cause `IsWhiteSpace` to be `false`; probably need a `[Theory]` taking a low number of whitespace examples as well as one or two `[Fact]`s that confirm a couple of combinations such as this one and what @NTaylorMullen suggested.

nit: whitespace is always singular, never "WhiteSpaces".  if you want to be really specific `IsWhiteSpace_True_AfterMultipleWhiteSpaceAdditions`.  (btw, "whitespace" is one word but don't mimic the first sentence in this paragraph.  capitalize the "S" because the BCL contains `IsNullOrWhiteSpace`)